Readd analysis class docs

nkanazawa1989 opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

This just readds link to the default analysis from the experiment class documentation. 
Note that default analysis options are not properly copied as reported in #561

### Details and comments

chriseclectic commented on qiskit_experiments/library/calibration/fine_amplitude.py:
```diff
@@ -35,6 +35,10 @@ class FineAmplitudeCal(BaseCalibrationExperiment, FineAmplitude):
         to it through the calibrations.
+
+    # section: see_also
+        qiskit_experiments.library.characterization.fine_amplitude.FineAmplitude
```
Will the analysis ref section be inherited from parent here? If not this should also have 
```python
# section: analysis_ref
        :py:class:`FineAmplitudeAnalysis`
```

nkanazawa1989 replied:
This is automatically handled in parser.
https://github.com/Qiskit/qiskit-experiments/blob/6812ac230f024ca748a5de56750aa81f93808703/docs/_ext/custom_styles/formatter.py#L95-L101

chriseclectic commented on qiskit_experiments/library/calibration/fine_amplitude.py:
```diff
@@ -143,7 +147,11 @@ def update_calibrations(self, experiment_data: ExperimentData):
 
 class FineXAmplitudeCal(FineAmplitudeCal):
-    """A calibration experiment to calibrate the amplitude of the X schedule."""
+    """A calibration experiment to calibrate the amplitude of the X schedule.
```
Same here and all the other subclasses of FineAmplitude and other subclassed experiments

nkanazawa1989 replied:
We can add analysis_ref here but this means if we update characterization analysis then one should note analysis ref is also here, which is not automatically detected. This is kind of soft reference through experiment class itself.

chriseclectic replied:
I'm fine with that. This is for benefit of people reading the API docs.

chriseclectic reviewed (commented):
Looks good, but one question: Do you need to add the `#section: analysis_ref` to subclassed experiments so it will show up in each subclasses documentation? (such as the spectroscopy, fine amplitude, drag ones) Or will it inherit that automatically?

nkanazawa1989 commented:
Added recursive `analysis_ref` reference to its superclass. The default analysis options (used to be in Experiment class doc) is moved and added to the Analysis class documentation. Now all calibration experiments automatically look up the associated characterization mix-in experiment documentation and port `analysis_ref` section from there. If one changed analysis class reference path in the superclass that will be propagated to all child classes, thus developer doesn't need to take care of inheritance of class for documentation.
